controllers/hbase_controller.go

```diff
@@ -564,9 +563,10 @@ func (r *HBaseReconciler) ensureStatefulSetPods(ctx context.Context, sts *appsv1
 func (r *HBaseReconciler) ensureStatefulSet(hb *hbasev1.HBase,
 	stsName, cmName types.NamespacedName, ss hbasev1.ServerSpec) (*appsv1.StatefulSet, bool, error) {
 	actual := &appsv1.StatefulSet{}
-	expected, expectedRevision := r.statefulSet(hb, stsName, cmName, ss)
 	if err := r.Get(context.TODO(), stsName, actual); err != nil {
 		if errors.IsNotFound(err) {
+			expected, _ := r.statefulSet(hb, stsName, cmName, ss, 0)
```

Comment:
what's the reason for this change?

Reply:
This is handling the case where statefulset is created for the first time, so for that the existing count would be 0 (best representative of "does not exist")